kafka/channel/pkg/apis/messaging/v1alpha1/kafka_channel_conversion.go

```diff
@@ -65,14 +65,21 @@ func (source *KafkaChannel) ConvertTo(ctx context.Context, obj apis.Convertible)
 			subscribableSpec.Subscribers = make([]eventingduckv1.SubscriberSpec, len(source.Spec.Subscribable.Subscribers))
 			for i, ss := range source.Spec.Subscribable.Subscribers {
 				delivery := &eventingduckv1.DeliverySpec{}
-				if err := ss.Delivery.ConvertTo(ctx, delivery); err != nil {
-					return err
+				if ss.Delivery != nil {
+					if err := ss.Delivery.ConvertTo(ctx, delivery); err != nil {
+						return err
+					}
+				}
+
+				var subscriberURI *apis.URL
+				if ss.SubscriberURI != nil {
+					subscriberURI = ss.SubscriberURI.DeepCopy()
 				}
```

Comment:
There is no need to check if `SubscriberURI` is nil since `DeepCopy` handles the nil case.

Reply:
Nice catch!

But, if that's ok, let's leave the cherry pick as is?

Reply:
I'm ok with that.